isazonov/src/test/java/com/evilcorp/hcf/HibernateCustomIntervalTests.java

```diff
@@ -0,0 +1,183 @@
+package com.evilcorp.hcf;
+
+import com.evilcorp.hcf.entities.Product;
+import com.evilcorp.hcf.repositories.interval.ProductIntervalRepository;
+import jakarta.annotation.PostConstruct;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Import(PostgresTestcontainersConfiguration.class)
+@SpringBootTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```

Comment:
Why using `@TestInstance(PER_CLASS)` here? If to keep data between test invocations then this is bad idea, use `@BeforeEach`

Reply:
I am using it to insert data once. Data is never changed, so it is safe. My tests are to demonstrate, that I can read that data with several different queries

Reply:
I know, I saw it. But this is in general a time bomb, and using `PER_CLASS` with data population can and probably would eventually lead to flaky tests. My point is - safe yourself and remove it.

Reply:
I don't agree with you on that, but I see no point in arguing. It's your decision )) .

Reply:
Tests are independent now